File: server/Models/RoomType.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Yes.Models;

public class RoomType
{
    [Key]
    public string Id { get; set; } = string.Empty;

    public virtual ICollection<Room> Rooms { get; set; } = [];

    [Required]
    [MaxLength(100)]
    public string Name { get; set; } = string.Empty;
    [MaxLength(255)]
    public string? Description { get; set; } = null;
    [Required]
    public int Price { get; set; } = 0;
}

File: frontend/types/RoomType.ts
export interface RoomType {
    id: string;
    name: string;
    description?: string | null;
    price: number;
}

export interface CreateRoomTypePayload {
    name: string;
    description?: string | null;
    price: number;
}

export interface UpdateRoomTypePayload {
    name?: string | null;
    description?: string | null;
    price?: number | null;
}